lcm-cmake/cpack.cmake

```diff
@@ -6,31 +6,29 @@ set(CPACK_GENERATOR TGZ STGZ)
 
 # Detect OS type, OS variant and target architecture
 if(UNIX)
-    if(APPLE)
-        set(OS_TYPE macos)
-    else()
-        set(OS_TYPE linux)
-
-        # Determine distribution
-        execute_process(COMMAND lsb_release -si
-            OUTPUT_VARIABLE LINUX_DISTRO
-            OUTPUT_STRIP_TRAILING_WHITESPACE
-        )
-
-        # Add CPack generator according to detected distribution
-        if(LINUX_DISTRO STREQUAL "Debian" OR LINUX_DISTRO STREQUAL "Ubuntu" OR LINUX_DISTRO STREQUAL "LinuxMint")
-            list(APPEND CPACK_GENERATOR DEB)
-        endif()
-    endif()
-
-    # Determine architecture
+   if(APPLE)
+    set(OS_TYPE macos)
+  else()
+    set(OS_TYPE linux)
+    # Determine distribution
+    file(READ /etc/lsb-release lsb_release)
+    string(REGEX REPLACE "^((.*ID|NAME)=)([a-zA-Z0-9]*).*" "\\3"  LINUX_DISTRO ${lsb_release})
```

Comment:
On my (Fedora) system, this produced `27`, which is definitely *not* the right answer. (This is because the above read the `VERSION_ID` field, when the correct field, assuming it should give the same result as `lsb_release -si`, comes from `NAME`.)